plaidml/op/op_test.cc

```diff
@@ -157,6 +156,112 @@ module {
 )#"));
 }
 
+TEST_F(OpTest, BroadcastNoOp) {
+  auto A = Placeholder(DType::FLOAT32, {3});
+  std::vector<int> rshape = {3};
+  std::vector<int> bdims = {0};
+  auto C = broadcast(A, rshape, bdims);
+  auto program = makeProgram("broadcast_nop", {C});
+
+  std::vector<float> A_input = {0, 1, 2};
+  checkProgram(program, {{A, A_input}}, {{C, A_input}});
+}
+
+TEST_F(OpTest, BroadcastNoOpLarge) {
+  auto A = Placeholder(DType::FLOAT32, {3, 4});
+  std::vector<int> rshape = {3, 4};
+  std::vector<int> bdims = {0, 1};
+  auto C = broadcast(A, rshape, bdims);
+  auto program = makeProgram("broadcast_nop_large", {C});
+
+  std::vector<float> A_input = {0, 1, 2, 3,  //
+                                0, 1, 2, 3,  //
+                                0, 1, 2, 3};
+  checkProgram(program, {{A, A_input}}, {{C, A_input}});
+}
+
+TEST_F(OpTest, BroadcastNumpy) {
+  auto A = Placeholder(DType::FLOAT32, {1, 3, 3});
+  std::vector<int> rshape = {1, 4, 3, 3};
+  std::vector<int> bdims = {0, 2, 3};
+  auto C = broadcast(A, rshape, bdims);
+  auto program = makeProgram("broadcast_numpy", {C});
+
+  std::vector<float> A_input = {0, 1, 2,  //
+                                0, 1, 2,  //
+                                0, 1, 2};
+  std::vector<float> C_output = {0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2,  //
+                                 0, 1, 2};
+}
+
+TEST_F(OpTest, BroadcastNumpy2) {
+  auto A = Placeholder(DType::FLOAT32, {3, 1});
+  std::vector<int> rshape = {3, 4};
+  std::vector<int> bdims = {0, 1};
+  auto C = broadcast(A, rshape, bdims);
+  auto program = makeProgram("broadcast_numpy_2", {C});
+
+  std::vector<float> A_input = {0, 1, 2};
+  std::vector<float> C_output = {0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2};
```

Comment:
I find it more readable to put in newlines for each 'row'.

Reply:
Sorry, must have missed this one while reformatting.